localstack/services/s3/provider.py

```diff
@@ -1150,6 +1161,109 @@ def get_object_attributes(
+        analytics_configurations: AnalyticsConfigurationList = (
+            store.bucket_analytics_configuration.get(bucket, [])
+        )
+        for analytics_configuration in analytics_configurations:
+            if analytics_configuration["Id"] == id:
+                return GetBucketAnalyticsConfigurationOutput(
+                    AnalyticsConfiguration=analytics_configuration
+                )
+        raise NoSuchConfiguration("The specified configuration does not exist.")
+
+    def list_bucket_analytics_configurations(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        continuation_token: Token = None,
+        expected_bucket_owner: AccountId = None,
+    ) -> ListBucketAnalyticsConfigurationsOutput:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store()
+        analytics_configurations: AnalyticsConfigurationList = (
+            store.bucket_analytics_configuration.get(bucket, [])
+        )
+        return ListBucketAnalyticsConfigurationsOutput(
+            IsTruncated=False, AnalyticsConfigurationList=analytics_configurations
+        )
+
+    def delete_bucket_analytics_configuration(
+        self,
+        context: RequestContext,
+        bucket: BucketName,
+        id: AnalyticsId,
+        expected_bucket_owner: AccountId = None,
+    ) -> None:
+        moto_backend = get_moto_s3_backend(context)
+        get_bucket_from_moto(moto_backend, bucket)
+
+        store = self.get_store()
+        analytics_configurations: AnalyticsConfigurationList = (
+            store.bucket_analytics_configuration.get(bucket, [])
+        )
+        for analytics_configuration in analytics_configurations:
+            if analytics_configuration["Id"] == id:
+                analytics_configurations.remove(analytics_configuration)
+                return
+        raise NoSuchConfiguration("The specified configuration does not exist.")
+
+
+def validate_analytics_configuration(analytics_configuration: AnalyticsConfiguration):
+    # TODO: add more concrete validation
+    # mapping of valid keys and whether they are required
+    required_keys = ["Id", "StorageClassAnalysis"]
+    optional_keys = ["Filter"]
+
+    errors = []
+    for key in analytics_configuration.keys():
+        # check if key is valid
+        if key not in required_keys + optional_keys:
+            errors.append(
+                f"Unknown parameter in AnalyticsConfiguration: \"{key}\", must be one of: {', '.join(required_keys + optional_keys)}"
+            )
+    for key in ["Id", "StorageClassAnalysis"]:
+        # check if required key is present
+        if key not in analytics_configuration.keys():
+            errors.append(f'Missing required parameter in AnalyticsConfiguration:: "{key}"')
+    if len(errors) > 0:
+        error_message = "\n".join(errors)
+        raise _create_invalid_argument_exc(error_message)
```

Comment:
this is a bit unsure for me now, because the exception raised is not the one that you snapshotted, as it was a boto client side validation. You might need to rewrite this once you have the proper server side exception from AWS.

Reply:
Thanks for pointing out. I've removed this validations.